cookbooks/tian_parameterization_kinematic_slab/coupled_two_phase_tian_parameterization_kinematic_slab.prm

```diff
@@ -0,0 +1,234 @@
+# This is a cookbook which simulates a portion of 2D subduction system with
+# a hydrated sediment layer, MORB layer, gabbro layer, and peridotite
+# mantle wedge. The layers dehydrate as they are subducted, with the solid-fluid
+# reactions governed by an approximation published by Tian et al., 2019.
+# The model simulates the upper 20 km of a slab subducting with a dip of 45 degrees
+# and a convergence rate of 5 cm/yr. The Cartesian box is rotated such that the
+# x-axis aligns with the slab surface, and the y-axis is orthogonal to the slab
+# surface, while gravity is imposed at an angle of 45 degrees. A surface pressure
+# of 1 GPa is imposed to account for the model simulating a portion of the slab
+# that is already subducted.
+# For stability, the slab first subducts for 100 kyr before the hydration
+# is added, and the slab surface temperature diffuses to the mantle over 10 km
+# perpendicular to the slab surface. From slab surface to the base of the model,
+# the composition is comprised of a 5km thick layer of sediment with 3 wt% water,
+# below is a 7km thick layer of MORB with 2 wt% water, and below that is an
+# 8 km thick layer of gabbro with 1 wt% water.
+
+set Adiabatic surface temperature              = 1600
+set Nonlinear solver scheme                    = iterated Advection and Stokes
+set Max nonlinear iterations                   = 25
+set Nonlinear solver tolerance                 = 1e-4
+set Dimension                                  = 2
+set End time                                   = 2e6
+set Surface pressure                           = 1e9
+set Maximum time step                          = 1000
+set Use operator splitting                     = true
+set Output directory                           = two-phase-tian
+
+subsection Checkpointing
+  set Steps between checkpoint = 200
+end
+
+subsection Discretization
+  # We choose relatively large values for the stabilization parameters:
+  # However, note that in an application model with a higher resolution,
+  # we would choose much smaller values for the stabilization parameters.
+  subsection Stabilization parameters
+    set beta  = 0.5
+    set cR    = 1
+  end
+end
+
+# Do 10 reaction time steps per advection time step.
+subsection Solver parameters
+  subsection Operator splitting parameters
+    set Reaction time step                     = 100
+    set Reaction time steps per advection step = 10
+  end
+
+  subsection Stokes solver parameters
+    set Linear solver tolerance             = 1e-10
+    set Number of cheap Stokes solver steps = 1000
+  end
+end
+
+subsection Compositional fields
```

Comment:
Unrelated to this subsection, but I think it would be interesting to compare the model results when using a continuous compositional field (default) versus a discontinuous DG field.